lib/practice_exercises.rb

```diff
@@ -1,13 +1,45 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 def remove_duplicates(list)
-  raise NotImplementedError, "Not implemented yet"
+  index = 0
+  prev_num = 0
+  list.length.times do
+    item = list[index]
+    if index == 0
+      prev_num = item
+      index += 1
+    else
+      if item == prev_num
+        list.delete_at(index)
+      else
+        index += 1
+        prev_num = item
+      end
+    end
+  end
+  return list
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(a * b) where a is the number of letters in the first word, and b is the number of words in the array
+# Space Complexity: O(1)
 def longest_prefix(strings)
-  raise NotImplementedError, "Not implemented yet"
+  longest_prefix = ""
+  num_letters = strings[0].length
+  
+  num_letters.times do |i|
+    status = true
+    check_letter = strings[0][i]
+    strings.each do |string|
+      if string[i] != check_letter
```

Comment:
If you find a letter that won't work you can stop here because any subsequent other letters can't be in the prefix.